src/js/components/HealthChecksComponent.jsx

```diff
@@ -184,10 +199,39 @@ var HealthChecksComponent = React.createClass({
                 }
                 fieldId={`${fieldsetId}.${i}.portIndex`}
                 label="Port Index"
-                value={row.portIndex}>
+                value={row.portIndex || 0}>
                 <input ref={`portIndex${i}`} {...numberInputAttributes} />
               </FormGroupComponent>
             </div>
+            <div className={portNumberClassSet}>
+              <FormGroupComponent
+                errorMessage={
+                  getErrorMessage(`${fieldsetId}.${i}.port`)
+                }
+                fieldId={`${fieldsetId}.${i}.port`}
+                label="Port Number"
+                value={row.port || 0}>
```

Comment:
@aldipower just one thing: while for `portIndex` the API actually defaults to 0, the same doesn't seem to be true for the newly introduced `port` (it defaults to none, apparently: https://github.com/mesosphere/marathon/blob/master/docs/docs/health-checks.md ) . Should we consider this?